StyleCop.Analyzers/StyleCop.Analyzers/Helpers/AccessLevelHelper.cs

```diff
@@ -234,6 +234,21 @@ internal static Accessibility GetDeclaredAccessibility(this BasePropertyDeclarat
             return declaredSymbol?.DeclaredAccessibility ?? Accessibility.NotApplicable;
         }
 
+        internal static Accessibility GetDeclaredAccessibility(this AccessorDeclarationSyntax syntax, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            Requires.NotNull(syntax, nameof(syntax));
+            Requires.NotNull(semanticModel, nameof(semanticModel));
+
+            AccessLevel accessLevel = GetAccessLevel(syntax.Modifiers);
+            if (accessLevel != AccessLevel.NotSpecified)
+            {
+                return accessLevel.ToAccessibility();
+            }
+
+            BasePropertyDeclarationSyntax propertyDefinition = syntax.FirstAncestorOrSelf<BasePropertyDeclarationSyntax>();
```

Comment:
:bulb: I would prefer to see `syntax.Parent.Parent` here, which is more efficient than the current check.